src/ad-bidders/prebid/utils/id-retriever.ts

```diff
@@ -1,20 +1,48 @@
 import { communicationService, eventsRepository } from '@ad-engine/communication';
-import { pbjsFactory, utils } from '@ad-engine/core';
+import { pbjsFactory, targetingService, UniversalStorage, utils } from '@ad-engine/core';
 
 export class IdRetriever {
 	private static _instance: IdRetriever;
-
+	private readonly ID_STRING_LENGTH = 16;
 	private readonly logGroup = 'bidder-id-retriever';
 	private pbjs: Pbjs;
 
+	private readonly ID_MAP: { [key: number]: (sources: PrebidEids[]) => string } = {
```

Comment:
```suggestion
	private readonly ID_MAP: Record<number, (sources: PrebidEids[]) => string>= {
```
it is recommended to use [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) over index signature